### PR TITLE
update docs for new release process

### PR DESCRIPTION
**Pull request by mroth**
~This reflects the state of things after #99 is merged, but opening PR now so we can begin looking at documentation manually.~

This reflects the state of things now that #99 is merged, for us to do the new flow but the very final release step manually. (Should still result in a cleaner and easier process with less steps, and we'll be able to resolve any CI/CD issues in the background and eventually automate that final step once we trust things to be fully reliable and everyone is up to speed on the flow.)

Rendered version of docs for ease of reading: https://github.com/openlawteam/openlaw-core/blob/docs-update/docs/ReleaseProcess.md

### CODE REVIEW

File: docs/ReleaseProcess.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+Note this documentation is for core project maintainers and all steps may not apply to open source contributors. If in doubt, just ask! We're friendly folks. :rainbow:
+
+## Submitting Changes :sparkles:
+
+New code changes should always be introduced via a Pull Request branch:
+
+1. Pull latest `master` branch then create new branch off from `master` and make the changes there.
+
+2. If added any new sbt tests, make sure they are all passing.
```
Comment (outkaj) [inline docs/ReleaseProcess.md]: by running `sbt test`

File: docs/ReleaseProcess.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+3. Submit PR for merging that new branch into `master`. GitHub Actions will automatically attempt a build and run tests, which you can see here: https://github.com/openlawteam/openlaw-core/actions.
+
+4. If the build and all checks are passing, the PR can be merged to master.
+
+## Releasing to BinTray :rocket:
+
+The `master` branch represents the current stable set of the code, and should be considered "release-able" at any point in time.
```
Comment (outkaj) [inline docs/ReleaseProcess.md]: should we also add - "Please don't merge anything into `master` until you feel it's production-ready."?

File: docs/ReleaseProcess.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+New _releases_ are cut from the `master` branch, and managed via GitHub Releases with [Semantic Versioning](https://semver.org/spec/v2.0.0.html) formatted tags. <!-- As a developer, this process is largely automated for you. -->
```
Comment (outkaj) [inline docs/ReleaseProcess.md]: should we add - "This process will be largely automated for you"?

File: docs/ReleaseProcess.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+To make a release of the current status of master, follow these steps:
```
Comment (outkaj) [inline docs/ReleaseProcess.md]: -> `master`

File: docs/ReleaseProcess.md (hunk 3 above)
Comment (outkaj) [reply docs/ReleaseProcess.md]: or just remove the above?

File: docs/ReleaseProcess.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+2. Confirm that the package was successfully published with the latest version number by [checking the BinTray listing](https://bintray.com/openlawos/openlaw-core/openlaw-core).
+
+#### Manually
+
+<!-- If for some reason, the publish to BinTray must be done manually in your local environment, you can do it via the following steps in an emergency: -->
+
+In the very near future, the following steps will be automated and done in CI, but for _right now_ we're doing them manually while we test:
+
+1. Switch to `master` and pull the latest. You may wish to also run `git fetch` to ensure you have the latest tags.
```
Comment (outkaj) [inline docs/ReleaseProcess.md]: -> You should also run `git fetch`

File: docs/ReleaseProcess.md (hunk 3 above)
Comment (mroth) [reply docs/ReleaseProcess.md]: Ah, I commented it out so it wouldn't show up when viewing the rendered markdown README, but could be pre-written and easily uncommented out once we "go live". I think I have a few things like that in there, hehe.

File: docs/ReleaseProcess.md (hunk 5 above)
Comment (mroth) [reply docs/ReleaseProcess.md]: I'd be interested to test this -- I _think_ `git pull` might automatically get any new tags if they are "reachable" from the commits you are pulling, e.g. fetch would only be required in the instance where you had the latest on your local copy before the tags were added to it. (But perhaps we should just ask people to always run fetch to be on the safe side.)

Comment (outkaj) [reply docs/ReleaseProcess.md]: You're probably right! 
I was more thinking of asking for completeness so we don't have to manually add tags again

Comment (mroth) [reply docs/ReleaseProcess.md]: I agree, better safe than sorry.  I'll make it very explicit when I do a revision pass on folks comments.

File: README.md [hunk 6]
```diff
@@ -1,26 +1,35 @@
...
 ```
 
-## Contributing 
+## Contributing
+
+Want to get involved? See detailed information about contributing [here](docs/CONTRIBUTING.md)!
+
+### Versioning
+
+Project versioning is automatically maintained from SemVer tags via `sbt-git`.
```
[the remark this answers is not in the document]
Comment (mroth) [reply README.md]: Good idea.  I did so in the Release Process docs but not here.  Will add.

File: docs/ReleaseProcess.md (hunk 2 above)
Comment (adridadou) [reply docs/ReleaseProcess.md]: I think it's more that we need to make sure that:
1. all the tests pass (no force merge there!) 
2. that we have done proper code review to make sure it is up to our standards. 
3. I think that the point in the document saying that it should be "release-able" at any point in time is a good way of summarizing all that.

In the future we should spend some time defining our definition of done for a task

**Pull request merged**